### DISCUSSION TITLE
Automatically checking the latest minecraft version and updating.

### DISCUSSION THREAD

**Discussion by Alddste** [itzg/docker-minecraft-server#2533] [Q&A] [answered]
Hello, I've been thinking about using the MODRINTH_PROJECTS in my docker-compose.yml. Howevever, I have two questions. I am looking to have as little human input as necessary, while keeping my quilt server up to date. I have a script which restarts my computer once a week. 

Is there a way to automatically check and update the quilt loader and minecraft version in the compose file itself or using a script?

Is it possible to have the compose file not re-download the same mod versions? This would save bandwidth and time every week after reboots.

I've thought about putting this in a script, but I wanted to check here first to see if there was any functionality already, or if anyone has found a method of checking the quilt/minecraft version automatically and updating it. Thanks!

**Comment by itzg** [comment, marked as answer]
> Hello, I've been thinking about using the MODRINTH_PROJECTS in my docker-compose.yml. Howevever, I have two questions. I am looking to have as little human input as necessary, while keeping my quilt server up to date. I have a script which restarts my computer once a week.
> 
> Is there a way to automatically check and update the quilt loader and minecraft version in the compose file itself or using a script?

If you leave both of those at their default of "latest", then that update check is already performed at container startup. Now if you're wondering something to run external to the running container to first check if there's an update, then that could be enhanced in the [mc-image-helper](https://github.com/itzg/mc-image-helper).

> 
> Is it possible to have the compose file not re-download the same mod versions? This would save bandwidth and time every week after reboots.

It should already be avoiding that. It does still have to contact Modrinth's API to lookup the latest version of each file or if the version was given, confirm the final filename of the mod/plugin.

> 
> I've thought about putting this in a script, but I wanted to check here first to see if there was any functionality already, or if anyone has found a method of checking the quilt/minecraft version automatically and updating it. Thanks!

If you want to do some research on the helper tool look at:
- https://github.com/itzg/mc-image-helper/tree/master/src/main/java/me/itzg/helpers/quilt
- https://github.com/itzg/mc-image-helper/blob/master/src/main/java/me/itzg/helpers/versions/ResolveMinecraftVersionCommand.java
